src/autoupdate.c

```diff
@@ -29,56 +29,76 @@
 
 #include "swupd.h"
 
-static void print_help(const char *name)
+static void print_help(void)
 {
 	print("Usage:\n");
-	print("   swupd %s [options]\n\n", basename((char *)name));
-	print("Help Options:\n");
-	print("   -h, --help              Show help options\n");
+	print("   swupd autoupdate [OPTION...]\n\n");
+
+	global_print_help();
+
+	print("Options:\n");
 	print("       --enable            enable autoupdates\n");
 	print("       --disable           disable autoupdates\n");
 	print("\n");
 }
 
-static int enable;
-static int disable;
+#define FLAG_ENABLE 2000
+#define FLAG_DISABLE 2001
+static bool cmdline_option_enable;
+static bool cmdline_option_disable;
 
 static const struct option prog_opts[] = {
 	{ "help", no_argument, 0, 'h' },
-	{ "enable", no_argument, &enable, 1 },
-	{ "disable", no_argument, &disable, 1 },
-	{ 0, 0, 0, 0 }
+	{ "enable", no_argument, 0, FLAG_ENABLE },
+	{ "disable", no_argument, 0, FLAG_DISABLE },
 };
 
-static bool parse_options(int argc, char **argv)
+static bool parse_opt(int opt, UNUSED_PARAM char *optarg)
 {
-	int opt;
-
-	while ((opt = getopt_long(argc, argv, "h", prog_opts, NULL)) != -1) {
-		switch (opt) {
-		case '?':
-			print_help(argv[0]);
-			exit(SWUPD_INVALID_OPTION);
-		case 'h':
-			print_help(argv[0]);
-			exit(EXIT_SUCCESS);
-		case 0: /* getopt_long has set the flag */
-			break;
-		default:
-			error("unrecognized option\n\n");
-			goto err;
+	switch (opt) {
+	case 'h':
+		print_help();
+		exit(EXIT_SUCCESS);
+	case FLAG_ENABLE:
```

Comment:
I told you this in a previous command but it is better to check flag restrictions within parse_options() as we do in the other commands

Reply:
Done